ide/project.dependency/src/org/netbeans/modules/project/dependency/Scope.java

```diff
@@ -86,9 +86,6 @@ public final boolean equals(Object obj) {
         if (obj == null) {
             return false;
         }
-        if (getClass() != obj.getClass()) {
```

Comment:
I am sorry, but this is wrong. I accept that two scopes with different implementation classes but the same name should be equal, but then please do something like:
```
if (!(obj instanceof Scope)) {
    return false;
}
```

I believe generally `equals` is expected to not crash on `CCE`s, even if an object of an unrelated type is passed in.

Reply:
Uh oh ... mea culpa. Of course.